app/models/concerns/dirty_associations.rb

```diff
@@ -14,16 +14,30 @@
 module DirtyAssociations
   extend ActiveSupport::Concern
 
+  included do
+    after_save :reset_dirty_cache_state
+  end
   module ClassMethods
     # usage:
     #   class Model
     #     dirty_has_many_associations :organizations, :locations
+
     def dirty_has_many_associations(*args)
+      @@dirty_associations ||= []
```

Comment:
don't use class variables like this, this would cause problems with class inheritance, note that @@vars are shared between parents and children classes. It's bettter to use [class_attribute](https://apidock.com/rails/v5.2.3/Class/class_attribute) or simply `@var` since this is a class scope already (note we're in module ClassMethods), simply replacing `@@` with `@` should do.

Reply:
I get a nil value when I try to access variables defined like this `@var` inside  dirty_has_many_associations function. @ares

Reply:
Yes. That makes sense. Made the change you suggested.